### PR TITLE
Issue #275: Fix unified and side-by-side diff view in the case hr_intraline = 1

### PR DESCRIPTION
**Pull request by futatuki**
The problems were:

1. `get_line` parameter for `DiffDescription._get_diff()` should be a function that returns a list of `str`, but we passed a function that returns a list of `bytes`.
2. In `ezt.py`,  the type of return value `_get_value()` should not be a generator object, but there were some cases they were.

I couldn't determine which shoud be fixed in 2. case,  ezt.py or its caller.  However as the former is easier, I'd taken it.

### CODE REVIEW

File: lib/viewvc.py [hunk 1]
```diff
@@ -4021,7 +4021,8 @@ def _fp_vclib_raw(self, left, right, fp, propname):
     def _content_lines(self, side, propname):
         f = self.request.repos.openfile(side.path_comp, side.rev, {})[0]
         try:
-            lines = f.readlines()
+            lines = [lin.decode(self.request.repos.encoding, 'surrogateescape')
+                     for lin in f.readlines()]
```
Comment (cmpilato) [inline lib/viewvc.py:4025]: is `lin` here meant to be `line`?

**Review by cmpilato — CHANGES_REQUESTED** [review]
Small variable name typo (I think?)

File: lib/viewvc.py (hunk 1 above)
Comment (futatuki) [reply lib/viewvc.py:4025]: Aaah, I had used '`l`' for the variable name at first, but flask8 blames "E741 ambiguous variable name 'l'".  Then I renamed it into '`lin`'. 

However, yes,  it is no mean that reduce only one character from full spell.  It's a same mistake of creat(2) :)

Comment (futatuki) [reply lib/viewvc.py]: Corrected in [1cad649](https://github.com/viewvc/viewvc/pull/298/commits/1cad6499cf480b950351dde81ffb9ae94cfd1cde)

**Review by cmpilato — APPROVED** [review]

**Pull request merged**